superset-frontend/src/components/Select/Select.tsx

```diff
@@ -577,7 +538,6 @@ const Select = ({
 
     if (filterOption) {
       const searchValue = search.trim().toLowerCase();
```

Comment:
We should probably remove trimming and case insensitivity treatment at this place since whitespaces and cases can be useful in search and ranking, too. For example, search "Le" may rank "Leah" first, but "Le " (with a space at the end) should rank things like "Le Monde" first.

Reply:
That's an interesting point. We also need to consider frequent use cases. If I search for "michael", I expect to get "Michael". We can tackle this in more depth when adding support to sophisticated backend search engines.